### DISCUSSION TITLE
Server-Side Events?

### DISCUSSION THREAD

**Discussion by Octodillo** [alephjs/aleph.js#439] [Q&A]
I'm trying out using Aleph for a new project I'm working on and I love it so far, but I'm struggling to utilize server-side events as part of the API. I know Oak supports SSE, but I can't figure out how to integrate it to Aleph's API framework if that's even possible. I'm very new to SSE so although I'm currently trying to implement it myself I'd be much happier with a built-in/library way of doing it. Does anyone know how that could work?

**Comment by sinyo-matu** [comment]
you may check [showcase_chat](https://github.com/denoland/showcase_chat) example.

The example use SSE inside this [handler](https://github.com/denoland/showcase_chat/blob/6f4a4dd8c1df8e65cb8d7e5d465965bc96970992/routes/api/connect.ts#L4)
you can do same thing in aleph
```typescript
import { Handlers, RouteConfig } from "$fresh/server.ts";
import { RoomChannel } from "@/communication/channel.ts";

export const handler: Handlers = {
  GET(_req, ctx) {
    const channel = new RoomChannel(+ctx.params.room);

    const stream = new ReadableStream({
      start: (controller) => {
        channel.onMessage((message) => {
          const body = `data: ${JSON.stringify(message)}\n\n`;
          controller.enqueue(body);
        });
      },
      cancel() {
        channel.close();
      },
    });

    return new Response(stream.pipeThrough(new TextEncoderStream()), {
      headers: { "content-type": "text/event-stream" },
    });
  },
};

export const config: RouteConfig = {
  routeOverride: "/api/connect/:room",
};
```
